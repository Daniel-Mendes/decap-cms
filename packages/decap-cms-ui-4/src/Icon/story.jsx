import React from 'react';
import styled from '@emotion/styled';
import { withKnobs, select } from '@storybook/addon-knobs';

import Icon, { iconComponents } from './Icon';

const StyledIcon = styled(Icon)`
  color: ${({ theme }) => theme.color.highEmphasis};
`;

export default {
  title: 'Components/Icon',
  decorators: [withKnobs],
};

export const _Icon = () => {
  return (
    <StyledIcon
      size={select('size', { xs: 'xs', sm: 'sm', md: null, lg: 'lg', xl: 'xl' }, null)}
      name={select(
        'name',
        {
          default: null,
          ...Object.keys(iconComponents).reduce((acc, key) => ({ ...acc, [key]: key }), {}),
        },
        null,
      )}
    />
  );
};
