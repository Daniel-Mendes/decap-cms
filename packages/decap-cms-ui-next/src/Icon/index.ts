export { default, iconComponents } from './Icon';
export type { IconName } from './Icon';
