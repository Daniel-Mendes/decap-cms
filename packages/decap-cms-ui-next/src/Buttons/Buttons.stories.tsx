import React from 'react';
import { userEvent, waitFor, within, expect, fn } from '@storybook/test';

import { Button, ButtonGroup, IconButton, AvatarButton, FileUploadButton } from '.';
import { iconComponents } from '../Icon';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Button> = {
  title: 'Components/Button',
  component: Button,
  args: {
    children: 'Button',
    onClick: fn(),
  },
};

export default meta;
type Story = StoryObj<typeof Button>;

// export const Default: Story = {
//   play: async ({ args, canvasElement, step }) => {
//     const canvas = within(canvasElement);

//     await step('Click button', async () => {
//       const button = canvas.getByRole('button');
//       await userEvent.click(button);
//     });

//     // assert
//     expect(args.children).toBe('Button');
//     await waitFor(() => expect(args.onClick).toHaveBeenCalled());
//   },
// };

export const Default: Story = {};

export const Neutral: Story = {
  args: {
    type: 'neutral',
  },
};

export const Success: Story = {
  args: {
    type: 'success',
  },
};

export const Danger: Story = {
  args: {
    type: 'danger',
  },
};

export const Disabled: Story = {
  args: {
    disabled: true,
  },
  play: async ({ args, canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Click button', async () => {
      const button = canvas.getByRole('button');
      await userEvent.click(button);

      expect(args.onClick).not.toHaveBeenCalled();
    });
  },
};

export const WithIcon: Story = {
  args: {
    icon: 'plus',
  },
};

export const WithMenu: Story = {
  args: {
    hasMenu: true,
  },
};

export const Transparent: Story = {
  args: {
    transparent: true,
  },
};

export const Solid: Story = {
  args: {
    variant: 'solid',
  },
};

export const Soft: Story = {
  args: {
    variant: 'soft',
  },
};

export const Small: Story = {
  args: {
    size: 'sm',
  },
};

export const Medium: Story = {
  args: {
    size: 'md',
  },
};

export const Large: Story = {
  args: {
    size: 'lg',
  },
};

export const ClickInteraction: Story = {
  play: async ({ args, canvasElement, step }) => {
    const canvas = within(canvasElement);

    await step('Click button', async () => {
      const button = canvas.getByRole('button');
      await userEvent.click(button);

      expect(args.onClick).toHaveBeenCalled();
    });
  },
};

// Render all variants of the button in a single story to take only one Image snapshot
export const ImageSnapshot: Story = {
  render: ({ args }) => (
    <ButtonGroup>
      <Button {...args}>Primary Button</Button>
      <Button {...args} type="success">
        Success Button
      </Button>
      <Button {...args} type="danger">
        Danger Button
      </Button>
      <Button {...args} type="neutral">
        Neutral Button
      </Button>
    </ButtonGroup>
  ),
};

// export function _Button(args) {
//   return <Button {...args}>Button</Button>;
// }

// _Button.argTypes = {
//   size: {
//     control: 'select',
//     options: ['sm', 'md', 'lg'],
//     mapping: {
//       md: null,
//     },
//     table: {
//       defaultValue: { summary: 'md' },
//     },
//   },
//   type: {
//     control: 'select',
//     options: ['primary', 'success', 'danger', 'neutral'],

//     table: {
//       defaultValue: { summary: 'primary' },
//     },
//   },
//   variant: {
//     control: 'select',
//     options: ['solid', 'soft'],
//     table: {
//       defaultValue: { summary: 'solid' },
//     },
//   },
//   disabled: {
//     control: 'boolean',
//     table: {
//       defaultValue: { summary: 'false' },
//     },
//   },
//   transparent: {
//     control: 'boolean',
//     table: {
//       defaultValue: { summary: 'false' },
//     },
//   },
//   icon: {
//     control: 'select',
//     options: {
//       default: null,
//       ...Object.keys(iconComponents).reduce((acc, key) => ({ ...acc, [key]: key }), {}),
//     },
//     table: {
//       defaultValue: { summary: 'null' },
//     },
//   },
// };

// _Button.args = {
//   type: 'primary',
//   size: 'md',
//   variant: 'solid',
//   disabled: false,
//   transparent: false,
//   icon: null,
//   onClick: fn(),
// };

// export function _ButtonGroup(args) {
//   return (
//     <ButtonGroup {...args}>
//       <Button>Button 1</Button>
//       <Button>Button 2</Button>
//       <Button primary type="success">
//         Primary Success Button
//       </Button>
//     </ButtonGroup>
//   );
// }

// _ButtonGroup.argTypes = {
//   direction: {
//     control: 'select',
//     options: ['horizontal', 'vertical'],
//     mapping: {
//       horizontal: null,
//     },
//   },
// };

// _ButtonGroup.args = {
//   direction: 'horizontal',
// };

// export function _IconButton(args) {
//   return <IconButton {...args} />;
// }

// _IconButton.argTypes = {
//   icon: {
//     control: 'select',
//     options: {
//       default: null,
//       ...Object.keys(iconComponents).reduce((acc, key) => ({ ...acc, [key]: key }), {}),
//     },
//   },
//   size: {
//     control: 'select',
//     options: ['sm', 'md', 'lg'],
//     mapping: {
//       md: null,
//     },
//   },
//   active: {
//     control: 'boolean',
//   },
// };

// _IconButton.args = {
//   icon: null,
//   size: 'md',
//   active: false,
//   onClick: fn(),
// };

// export function _AvatarButton(args) {
//   return <AvatarButton {...args} />;
// }

// _AvatarButton.argTypes = {
//   src: {
//     control: 'text',
//   },
//   size: {
//     control: 'select',
//     options: ['sm', 'md', 'lg'],
//     mapping: {
//       md: null,
//     },
//   },
//   active: {
//     control: 'boolean',
//   },
// };

// _AvatarButton.args = {
//   src: 'https://randomuser.me/api/portraits/men/94.jpg',
//   size: 'md',
//   active: false,
//   onClick: fn(),
// };

// export function _FileUploadButton(args) {
//   return <FileUploadButton {...args} />;
// }

// _FileUploadButton.argTypes = {
//   label: {
//     control: 'text',
//   },
//   accept: {
//     control: 'select',
//     options: ['audio/*', 'image/*', 'video/*', '*/*'],
//   },
//   size: {
//     control: 'select',
//     options: ['sm', 'md', 'lg'],
//     mapping: {
//       md: null,
//     },
//   },
//   disabled: {
//     control: 'boolean',
//     table: {
//       defaultValue: { summary: 'false' },
//     },
//   },
// };

// _FileUploadButton.args = {
//   label: 'Upload File',
//   accept: 'image/*',
//   size: 'md',
//   disabled: false,
//   onChange: fn(),
// };
