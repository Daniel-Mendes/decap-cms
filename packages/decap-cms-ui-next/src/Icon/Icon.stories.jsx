import React from 'react';

import Icon, { iconComponents } from './Icon';

export default {
  title: 'Components/Icon',
  component: Icon,
  argTypes: {
    name: {
      control: 'select',
      options: {
        default: null,
        ...Object.keys(iconComponents).reduce((acc, key) => ({ ...acc, [key]: key }), {}),
      },
      table: {
        defaultValue: { summary: 'null' },
      },
    },
    size: {
      control: 'select',
      options: ['xs', 'sm', 'md', 'lg', 'xl'],
      mapping: {
        md: null,
      },
      table: {
        defaultValue: { summary: 'md' },
      },
    },
  },
  args: {
    name: null,
    size: 'md',
  },
};

export function _Icon(args) {
  return <Icon {...args} />;
}
