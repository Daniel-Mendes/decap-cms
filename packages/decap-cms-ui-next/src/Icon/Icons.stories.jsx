import React from 'react';

import Icon, { iconComponents } from './Icon';

export default {
  title: 'Foundations/Icons Library',
  component: Icon,
  parameters: {
    options: {
      showPanel: false,
    },
  },
};

export function IconsLibrary() {
  return (
    <div
      style={{
        display: 'flex',
        flexDirection: 'column',
        justifyContent: 'center',
        margin: '0 10rem',
      }}
    >
      <h1>Icons Library</h1>

      <div
        style={{
          display: 'flex',
          flexWrap: 'wrap',
          flexDirection: 'row',
          justifyContent: 'center',
          alignItems: 'center',
        }}
      >
        {Object.keys(iconComponents).map(name => (
          <div
            key={name}
            style={{
              flex: '1 1 25%',
              display: 'flex',
              flexDirection: 'row',
              alignItems: 'flex-end',
              gap: '1rem',
            }}
          >
            <Icon name={name} size="md" />
            <span style={{ marginTop: 4 }}>{name}</span>
          </div>
        ))}
      </div>
    </div>
  );
}
