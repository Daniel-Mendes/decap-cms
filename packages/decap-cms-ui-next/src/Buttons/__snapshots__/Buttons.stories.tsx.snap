// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components/Button ClickInteraction play-test 1`] = `"<button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Danger smoke-test 1`] = `"<button type="danger" class="css-guei5i-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Default smoke-test 1`] = `"<button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Disabled play-test 1`] = `"<button type="primary" class="css-64dfqu-StyledButton-StyledButton e9pg3mh2" disabled="">Button</button>"`;

exports[`Components/Button ImageSnapshot smoke-test 1`] = `"<div class="css-o7p2vo-ButtonGroup e1v5bfbh0"><button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2">Primary Button</button><button type="success" class="css-1mqce4o-StyledButton-StyledButton e9pg3mh2">Success Button</button><button type="danger" class="css-guei5i-StyledButton-StyledButton e9pg3mh2">Danger Button</button><button type="neutral" class="css-1a8ppn3-StyledButton-StyledButton e9pg3mh2">Neutral Button</button></div>"`;

exports[`Components/Button Large smoke-test 1`] = `"<button type="primary" class="css-e2gzmo-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Medium smoke-test 1`] = `"<button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Neutral smoke-test 1`] = `"<button type="neutral" class="css-1a8ppn3-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Small smoke-test 1`] = `"<button type="primary" class="css-1rx6t3g-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Soft smoke-test 1`] = `"<button type="primary" class="css-1c08an4-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Solid smoke-test 1`] = `"<button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Success smoke-test 1`] = `"<button type="success" class="css-1mqce4o-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button Transparent smoke-test 1`] = `"<button type="primary" class="css-17sq55t-StyledButton-StyledButton e9pg3mh2">Button</button>"`;

exports[`Components/Button WithIcon smoke-test 1`] = `"<button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2"><svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" class="css-1yl0z6r-StyledIcon e9pg3mh0" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><line x1="12" y1="5" x2="12" y2="19"></line><line x1="5" y1="12" x2="19" y2="12"></line></svg>Button</button>"`;

exports[`Components/Button WithMenu smoke-test 1`] = `"<button type="primary" class="css-1xculjk-StyledButton-StyledButton e9pg3mh2">Button<svg xmlns="http://www.w3.org/2000/svg" width="20" height="20" class="css-1ebphsb-StyledMenuIcon e9pg3mh1" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"><polyline points="6 9 12 15 18 9"></polyline></svg></button>"`;
